### ISSUE TITLE
Ready

### ISSUE THREAD

**Issue by JohnKhandygo** [SergeyChernonog/XORcipher#1]
Вам осталось чуть-чуть исправить в первом задании. После этого жду отмашки, когда можно будет проверять. Сейчас у Вас нет тестов, а без них я проверку начинать не буду.

**Comment by SergWh** [comment]
Добавил тесты для функции кодирования

**Comment by JohnKhandygo** [comment]
Написал несколько мелких замечаний. У Вас не должно возникнуть проблем с их исправлением. Подумайте пока над третьим заданием. Здесь есть несколько вариантов:
1. По плану третье задание заключается в разработке графического приложения на JavaFX/Swing. Как и раньше я могу выдать Вам одно из готовых заданий, либо Вы можете предложить свое. 
2. Я могу выдать Вам задание на разработку веб приложения. Здесь основной упор будет на то, что Вам придется самостоятельно (я дам стартовый набор материалов) изучить несколько технологий и реализовать приложение. С точки зрения логики приложения ничего супер сложного там не будет. Задание будет предусматривать набор опциональных пунктов (например GUI, работа с базой данных), чтобы Вы могли выбрать наиболее интересные Вам технологии.
3. Что-то совсем другое. То есть какие-то Ваши идеи. Возможно Android или какие-то распределенные системы. В этом пути мне нужна какая-то вводная идея от Вам, чтобы на основе этого можно было сформулировать полноценное задание. 

Первый вариант наиболее безопасен с той точки зрения, что у Вас наверняка будут спрашивать что-то по графике на экзамене (раз это входит в программу). Второй и третий варианты на мой взгляд более интересные, позволяют Вам изучить современные востребованные технологии.

Напишите о своем решении как можно быстрее, чтобы не затягивать с выдачей третьего задания.

**Comment by SergWh** [comment]
Можно ли вдвоем делать задание ?
Одногруппник предложил идею для совместного проекта

**Comment by JohnKhandygo** [comment]
Не знаю, мне надо посоветоваться с другими преподавателями. Что за идея?

**Comment by SergWh** [comment]
https://habrahabr.ru/post/336330/
что-то подобное, только с использованием генетического алгоритма при появлении новых особей

**Comment by JohnKhandygo** [comment]
Честно говоря, не выглядит как задание для двоих. Я обсужу с другими преподавалями и дам Вам более точный ответ. Одногруппник, с которыми Вы хотите сотрудничать, это кто?

**Comment by SergWh** [comment]
Никита Киселёв

**Comment by JohnKhandygo** [comment]
В общем мы решили, что это точно слишком просто для двоих.

**Comment by JohnKhandygo** [comment]
Можете подумать как усложнить задачу. Например, ввести больше разных видов, моделирование источников пищи для травоядных, может смену времён года.  

Если получится достаточно сложная задача, то да, можно делать ее вдвоем. Только учтите, что Киселев ещё даже не выложил второе задание.

**Comment by SergWh** [comment]
Изначально идея была в том, чтобы поведение животного задавалось не просто проверкой окружающих условий, а небольшой нейронной сетью. Тогда генотипом при скрещивании были бы коэффициенты в сетях двух особей.
И мы думали о добавлении времен года и разном ландшафте

**Comment by JohnKhandygo** [comment]
Так, я все обдумывал Ваше предложение. В принципе я не против, но есть одно но: мне сложно сказать, как правильно организовать скрещивание для таких нейронных сетей, поэтому если в этой части что-то не будет работать, я не смогу как-то адекватно помочь. 

В понедельник жду от вас обоих концепцию задания, где будет более менее точно описано, что Вы готовы сделать, как это будет работать и как примерно будет реализован генетический алгоритм (я имею ввиду сами операции над нейронными сетями, без кода). Если задание получится достаточно  емким для давоих и мы точно определим необходимую функциональность приложения, то окей, можно делать. 

Такой вариант устроит?

**Comment by SergWh** [comment]
Да, конечно

Еще вопрос, если с этим заданием не получится, можно ли в качестве плана Б написать что-то?

**Comment by JohnKhandygo** [comment]
Конечно. Можете придумать собственный вариант Б, можно взять что-то из того, что я предлагал выше.

**Comment by JohnKhandygo** [comment]
Второе задание зачтено.

**Comment by SergWh** [comment]
Не уверен, что сейчас с совместным проектом, пока похоже, что не успеваем
Вы разрешили сделать еще и "план Б" https://github.com/SergeyChernonog/GO_game
Репозиторий с реализацией игры атари-го

**Comment by JohnKhandygo** [comment]
Сложный вопрос. С одной стороны, я ничего не имею против, с другой Вы, во-первых, подставляете своего коллегу (в одиночку-то он с еще меньшей вероятностью успеет сделать), а во-вторых, можно было бы и раньше написать: менять задание в последний день --- такая себе идея. Я посоветуюсь с другими преподавателями и (надеюсь) завтра сообщу Вам решение по данному вопросу.

**Comment by SergWh** [comment]
Ну, если честно, проблема в том , что пока почти весь (по сути весь) совместный проект написан Никитой
Я продолжу его делать, но если дедлайн в вс, то может получиться, что мой вклад в задание будет минимальным
А с учетом того, что я уже сдал экзамен (получается без допуска к экзамену), мое положение сейчас совсем неясное

**Comment by DeKinci** [comment]
>подставляете своего коллегу

Мне не критично. Экзамен я сдал, для зачета написать немного не сложно. Так вышло, что неделю, которую я планировал потратить на проект я потратил на подготовку к собеседованию и теперь приходится писать все в последней вечер.

**Comment by JohnKhandygo** [comment]
Допуск к экзамену у Вас у обоих итак есть. Оценка за практику отдельно ставится в зачетку как оценка за курсовую работу. Касательно внутреннего разделения труда между вами: никакого контроля за эти изначально не было предусмотрено, поэтому пока нет претензий от кого-то одного из вас, то тут и говорить не о чем. Поэтому положение вполне себе ясное.

**Comment by JohnKhandygo** [comment]
@SergeyChernonog, коллективное решение по Вашему вопросу примерно следующее: никому не понравилось, что Вы пытаетесь провернуть этот трюк в последний момент, кроме того, не очень понятно, почему (конфликт между Вами отсутствует, насколько я понял). Поэтому доделывайте то задание, на которое изначально договаривались.

**Comment by SergWh** [comment]
ок
